PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = 'account.journal'
-    _inherit = 'account.move'
-    _inherit = 'ir.actions.report'
-    _inherit = 'ir.attachment'

Total Changes: 12

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/account_edi/models/account_edi_document.py
+++ b/addons/account_edi/models/account_edi_document.py
@@ -15,7 +15,6 @@ DEFAULT_BLOCKING_LEVEL = 'error'
 
 
 class AccountEdiDocument(models.Model):
-    _name = 'account.edi.document'
     _description = 'Electronic Document for an account.move'
 
     # == Stored fields ==

--- a/addons/account_edi/models/account_edi_format.py
+++ b/addons/account_edi/models/account_edi_format.py
@@ -19,7 +19,6 @@ _logger = logging.getLogger(__name__)
 
 
 class AccountEdiFormat(models.Model):
-    _name = 'account.edi.format'
     _description = 'EDI format'
 
     name = fields.Char()

--- a/addons/account_edi/models/account_journal.py
+++ b/addons/account_edi/models/account_journal.py
@@ -8,7 +8,7 @@ from collections import defaultdict
 
 
 class AccountJournal(models.Model):
-    _inherit = 'account.journal'
+    _inherit = ['account.journal']
 
     edi_format_ids = fields.Many2many(comodel_name='account.edi.format',
                                       string='Electronic invoicing',

--- a/addons/account_edi/models/account_move.py
+++ b/addons/account_edi/models/account_move.py
@@ -10,7 +10,7 @@ from odoo.exceptions import UserError
 
 
 class AccountMove(models.Model):
-    _inherit = 'account.move'
+    _inherit = ['account.move']
 
     edi_document_ids = fields.One2many(
         comodel_name='account.edi.document',

--- a/addons/account_edi/models/account_move_send.py
+++ b/addons/account_edi/models/account_move_send.py
@@ -2,7 +2,7 @@ from odoo import api, models
 
 
 class AccountMoveSend(models.AbstractModel):
-    _inherit = 'account.move.send'
+    _inherit = ['account.move.send']
 
     @api.model
     def _get_mail_attachment_from_doc(self, doc):

--- a/addons/account_edi/models/ir_actions_report.py
+++ b/addons/account_edi/models/ir_actions_report.py
@@ -7,7 +7,7 @@ from odoo.tools.pdf import OdooPdfFileReader, OdooPdfFileWriter
 
 
 class IrActionsReport(models.Model):
-    _inherit = 'ir.actions.report'
+    _inherit = ['ir.actions.report']
 
     def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
         # EXTENDS base

--- a/addons/account_edi/models/ir_attachment.py
+++ b/addons/account_edi/models/ir_attachment.py
@@ -5,7 +5,7 @@ from odoo.exceptions import UserError
 
 
 class IrAttachment(models.Model):
-    _inherit = 'ir.attachment'
+    _inherit = ['ir.attachment']
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_government_document(self):
